spring/src/main/java/org/axonframework/spring/authorization/MessageAuthorizationDispatchInterceptor.java

```diff
@@ -49,9 +70,13 @@ public T handle(@Nonnull T message) {
         }
 
         logger.debug("Adding message metadata for username & authorities.");
-        Map<String, Object> authenticationDetails = new java.util.HashMap<>();
-        authenticationDetails.put("username", authentication.getPrincipal());
-        authenticationDetails.put("authorities", authentication.getAuthorities());
+        Map<String, String> authenticationDetails = new java.util.HashMap<>();
+        authenticationDetails.put("username", converter.convert(authentication.getPrincipal(), String.class));
+        String authorities = authentication.getAuthorities()
+                                           .stream()
+                                           .map(GrantedAuthority::getAuthority)
+                                           .collect(Collectors.joining(","));
+        authenticationDetails.put("authorities", authorities);
```

Comment:
Looking at that, I'm still not sure if the change it's not a step back.

Reply:
Ah, yea, I hit this part, and figured, we don't need the entire object. Checking the `GrantedAuthority` implementations, this seemed sufficient.
Given how recent this component is, I don't expect a lot of people "breaking" because only part of the `GrantedAuthority` is in the meta data.

If you want me to revert this when the new converter methods are present, just let me know.